viadot/sources/sap_rfc.py

```diff
@@ -84,6 +85,103 @@ def trim_where(where: str) -> Tuple[str, OrderedDictType[str, str]]:
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        chunk (int): The part number in which a number of SAP table columns have been split.
+        fields (List[str]): A list with the names of the columns in a chunk.
+
+    Returns:
+        Union[int, np.array, bool]: A tuple with the parameters "row_index", "data_raw", and a new
+            boolean variable "cont" to indicate when the for loop has to be restarted.
+    """
+    cont = False
+    if row_index == 0:
+        row_index = data_raw.shape[0]
+        if row_index == 0:
+            logger.warning(
+                f"Empty output was generated for chunk {chunk} in columns {fields}."
+            )
+            chunk += 1
+            cont = True
+    elif data_raw.shape[0] != row_index:
+        data_raw = data_raw[:row_index]
+        logger.warning(
+            f"New rows were generated during the execution of the script. The table is truncated to the number of rows for the first chunk"
+        )
+
+    return row_index, data_raw, cont
+
+
+def catch_extra_separators(
+    data_raw: np.array, record_key: str, sep: str, fields: List[str], replacement: str
+) -> np.array:
+    """Function to replace extra separators in every row of the
+
+    Args:
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        record_key (str): Key word to extract the data from the numpy array "data_raw".
+        sep (str): Which separator to use when querying SAP.
+        fields (List[str]): A list with the names of the columns in a chunk.
+        replacement (str): In case of sep is on a columns, set up a new character to replace
+            inside the string to avoid flow breakdowns.
+
+    Returns:
+        np.array: The argument "data_raw" with no extra delimiters.
+    """
+
+    # remove scape characters from data_raw ("\t")
+    for n, r in enumerate(data_raw):
+        if "\t" in r[record_key]:
+            data_raw[n][record_key] = r[record_key].replace("\t", " ")
+
+    # first we identify where the data has an extra separator in text columns.
+    sep_counts = np.array([], dtype=int)
+    for row in data_raw:
+        sep_counts = np.append(sep_counts, row[record_key].count(f"{sep}"))
+
+    no_sep_index = np.argwhere(sep_counts != len(fields) - 1)
+    no_sep_index = no_sep_index.reshape(
+        len(no_sep_index),
+    )
+    sep_index = np.argwhere(sep_counts == len(fields) - 1)
+    sep_index = sep_index.reshape(
+        len(sep_index),
+    )
+    # indentifying "good" rows we obtain the index of separatos positions.
+    pos_sep_index = np.array([], dtype=int)
+    for data in data_raw[sep_index]:
+        pos_sep_index = np.append(
+            pos_sep_index,
+            np.where(np.array([*data[record_key]]) == f"{sep}"),
+        )
+    pos_sep_index = np.unique(pos_sep_index)
+
+    # in rows with an extra separator, we replace them by another character: "-" by default
```

Comment:
The function name is `catch_extra_separator`. Maybe it is a good idea to create new function because this one seems to be complex. You can extract the `replace_separator_in_data` code block from this function and create a new one.

Reply:
I think all the code in that function should stay together because everything is related. Maybe I don't understand exactly what you want to say

Reply:
I meant to separate this part of the code and instead of a block of code call a new function here. This part of the functionality would be named for example `replace_separator_in_data` (or in a similar way). Then `catch_extra_separators` will be shorter and (I hope) easier to read. Creating smaller functions can help us, for example, catch errors faster in the future.
What do you think?

Reply:
done!